huaweicloud/services/sdrs/resource_huaweicloud_sdrs_drill.go

```diff
@@ -0,0 +1,172 @@
+	if err != nil {
+		return diag.Errorf("error creating SDRS DR drill: %s", err)
+	}
+
+	createTimeoutSec := int(d.Timeout(schema.TimeoutCreate).Seconds())
+	if err = drill.WaitForJobSuccess(client, createTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	drillID, err := drill.GetJobEntity(client, n.JobID, "disaster_recovery_drill_id")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(drillID.(string))
+	return resourceDrillRead(ctx, d, meta)
+}
+
+func resourceDrillRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	n, err := drill.Get(client, d.Id()).Extract()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.1902" {
+				// `SDRS.1902` means DR drill not found
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "error retrieving SDRS DR drill")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	mErr := multierror.Append(
+		nil,
+		d.Set("name", n.Name),
+		d.Set("group_id", n.GroupID),
+		d.Set("drill_vpc_id", n.DrillVpcID),
+		d.Set("status", n.Status),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceDrillUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	if d.HasChange("name") {
+		updateOpts := drill.UpdateOpts{
+			Name: d.Get("name").(string),
+		}
+		_, err = drill.Update(client, d.Id(), updateOpts).Extract()
+		if err != nil {
+			return diag.Errorf("error updating SDRS DR drill name, %s", err)
+		}
+	}
+	return resourceDrillRead(ctx, d, meta)
+}
+
+func resourceDrillDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	n, err := drill.Delete(client, d.Id()).ExtractJobResponse()
+	if err != nil {
+		if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+			if resp, pErr := common.ParseErrorMsg(errCode.Body); pErr == nil && resp.ErrorCode == "SDRS.1902" {
+				// `SDRS.1902` means DR drill not found
+				return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "error deleting SDRS DR drill")
+			}
+		}
+		return diag.FromErr(err)
+	}
+
+	deleteTimeoutSec := int(d.Timeout(schema.TimeoutDelete).Seconds())
+	if err := drill.WaitForJobSuccess(client, deleteTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
```

Comment:
```suggestion
	d.SetId("")
	return nil
```

Reply:
The deletion method will automatically set the ID to empty internally.